### ISSUE TITLE
Extractor extracts all backend services and name values

### ISSUE THREAD

**Issue by wsilveiranz** [Azure/azure-api-management-devops-resource-kit#377]
When running the extractor to extract a single API, the extractor generates all name values and all backend services, instead of just the items related to the API. It also extracts all the name values in clear text, even if the value is marked as a secret. Ideally:

1. 1. Only backend associated with the extracted API should be generated.
2. 2. Only name values associated with the extracted API should be generated (I've noticed that in the portal, trying to delete a name value associated to a policy generates an error, so seems like some background code to leverage.
3. Secret Name Values are sanitized when extracted. I would prefer to have to load them add values again in the parameter files at a later stage, then load the file with secrets to a repo by mistake.

I hope this helps explaining the issue. Happy to expand on that.

Cheers, Wagner.

**Comment by KevinHale-Bentley** [comment]
@wsilveiranz , I agree with you on this.  I have a work-around for my needs, based on Powershell tooling that I had already written, but it is not an ideal solution.  I thought about taking a stab at this in the Extractor code and submitting a PR, but I have to find time first.

**Comment by joeyeng** [comment]
This actually prevents me from using the extractor, because we have so many named values and backends that the command times out before it can finish extracting everything.

**Comment by joeyeng** [comment]
Hi @lucashuet93 I noticed in commit 82edf486 you removed the ability to limit extraction to referenced backends. Just curious what the reasoning behind this was, and if it's okay to uncomment it? Thanks.

**Comment by joeyeng** [comment]
If I understand correctly, the issue is if the policies are split out into separate xml files (if policyXMLBaseUrl param is set), so then you'd either have to load the contents of the xml files to see if the backend was referenced or you'd have to refactor the APIExtractor so it has that logic when it pulls the contents of the policy (probably the preferred approach perf wise). Does that sound about right?

**Comment by joeyeng** [comment]
#151

**Comment by adrianhall** [comment]
For named values, we are doing #151 in 1.0.0-beta.3.  For backends, we will be doing it as a follow-on.  To prevent duplication, I am closing this issue.  Thank you for your time in reporting it.
